src/js/player.js

```diff
@@ -2218,27 +2219,23 @@ class Player extends Component {
    *         The current video source when getting
    */
   src(source) {
-    if (source === undefined) {
-      return this.cache_.src;
-    }
-
-    this.changingSrc_ = true;
-
     let src = source;
 
-    if (Array.isArray(source)) {
-      this.cache_.sources = source;
-      src = source[0];
-    } else if (typeof source === 'string') {
-      src = {
-        src: source
-      };
+    // filter out invalid sources and turn our source into
+    // an array of source objects
+    src = filterSource(src);
 
-      this.cache_.sources = [src];
+    // no valid sources, return the current source
+    if (!src.length) {
+      return this.cache_.src;
     }
 
-    this.cache_.source = src;
+    this.cache_.sources = src;
+    src = src[0];
+
+    this.changingSrc_ = true;
 
+    this.cache_.source = src;
     this.currentType_ = src.type;
```

Comment:
I am wondering two things now:
1. Do we need to change `this.cache_.source`, `this.cache_.src`, and `this.cache_.type` in the `middleware.setSource` callback and also above it? or an we just do it inside?
2. Should we even cache `this.cache_.src` and `this.cache_.type` or should we just cache `this.cache_.source` and use that when we need to get the `type` or the `src` url